src/components/Pokelist/index.tsx

```diff
@@ -11,22 +10,27 @@ import {
 } from "@chakra-ui/react";
 
 import PokeCard from "../PokeCard";
-import { AllPokemonsResponse, PokecardProps } from "../../types";
+import { PokecardProps } from "../../types";
 import CenteredWrapper from "../CenteredWrapper";
+import useFetchPokemons from "../../hooks/useFetchPokemons";
 
 const DEFAULT_OFFSET = 20;
 const Pokelist: FC = () => {
   const [offset, setOffset] = useState<number>(0);
-  const fetchProjects = (offset = 0) =>
-    fetch(`https://pokeapi.co/api/v2/pokemon/?limit=20&offset=${offset}`).then(
-      (res) => res.json()
-    );
-  const { isLoading, error, data, isFetching } = useQuery<
-    AllPokemonsResponse,
-    Error
-  >(["allPokemons", offset], () => fetchProjects(offset), {
-    keepPreviousData: true,
-  });
+  const [currentPage, setCurrentPage] = useState<number>(0);
```

Comment:
Nice man! I like how you keep a state with the page only. Maybe you can ditch the offset state var and calculate it?